### PR TITLE
Refactor common elements to base/ directory

### PR DESCRIPTION
**Pull request by ivan-c**
- Split `dev/` directory into `base/` and `dev/` overrides
- Store secrets in service-specific `env_file`s, instead of global `.env`
- Add rough README
- Base `LOGSERVER_URL` on `BASE_DOMAIN`

#### Todo
- [ ] Comprehensive README

### CODE REVIEW

**Review by pbugni — APPROVED** [review]
See comments for minor concerns.
A ton of change, can't wait for finished product.
Thanks for all your organization work to attempt to keep this beast manageable!

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "dev/freestanding/logs/logserver"]
-	path = dev/freestanding/logs/logserver
+	path = base/freestanding/logs/logserver
```
[the remark this answers is not in the document]
Comment (ivan-c) [reply .gitmodules:2]: I used `git mv` to move the submodule, which is supposed to "do the right thing".

[It sounds like line 1 is just a name/reference, and isn't evaluated as a path](https://stackoverflow.com/questions/4604486/how-do-i-move-an-existing-git-submodule-within-a-git-repository/4607346#4607346).

I'll go ahead and change it anyway so it's less confusing, but be prepared to re-clone cosri-environments if necessary :)

File: dev/freestanding/femr/README.md [hunk 2]
```diff
@@ -1,24 +1,18 @@
...
 
-```
-cd ./cosri-environments/dev
-docker-compose up
-```
+Copy the .env file default
+    cp default.env .env
 
-Finally, navigate to the client: https://dashboard.localtest.me
+Modify each newly copied env file as necessary. Lines that are not commented-out are required, commented lines are optional.
```
[the remark this answers is not in the document]
Comment (ivan-c) [reply dev/freestanding/femr/README.md:16]: A little too implicit, but would be a bit much to add required/optional to every config value

File: dev/freestanding/femr/docker-compose.yaml [hunk 3]
```diff
@@ -1,186 +1,49 @@
...
-      - "traefik.enable=true"
-      - "traefik.http.routers.dashboard-${COMPOSE_PROJECT_NAME}.rule=Host(`dashboard.${BASE_DOMAIN:-localtest.me}`)"
-      - "traefik.http.routers.dashboard-${COMPOSE_PROJECT_NAME}.entrypoints=websecure"
-      - "traefik.http.routers.dashboard-${COMPOSE_PROJECT_NAME}.tls=true"
-      - "traefik.http.routers.dashboard-${COMPOSE_PROJECT_NAME}.tls.certresolver=letsencrypt"
-    networks:
-      - ingress
-      - internal
+    env_file:
+      - ../../../dev/freestanding/femr/dashboard.env
```
[the remark this answers is not in the document]
Comment (ivan-c) [reply dev/freestanding/femr/docker-compose.yaml:5]: This was a surprise to me too: all paths are relative to the first docker-compose file (ie in `base/`)

>Tracking which fragment of a service is relative to which path is difficult and confusing, so to keep paths easier to understand, all paths must be defined relative to the base file.

[Understanding multiple Compose files](https://docs.docker.com/compose/extends/#adding-and-overriding-configuration)

**Pull request merged**